src/test/java/com/thangle/api/user/UserControllerTest.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.http.MediaType;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+import java.util.UUID;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import static com.thangle.fakes.UserFakes.buildUsers;
+import static com.thangle.fakes.UserFakes.buildUser;
+import static com.thangle.api.user.UserDTOMapper.toUserDTO;
+
+@WebMvcTest(UserController.class)
+@AutoConfigureMockMvc
+public class UserControllerTest {
+    private static final String BASE_URL = "/api/v1/users";
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private UserService userService;
+
+    @Test
+    public void shouldFindAll_OK() throws Exception {
+        final var users = buildUsers();
+
+        when(userService.findAll()).thenReturn(users);
+
+        mvc.perform(MockMvcRequestBuilders.get(BASE_URL))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()").value(users.size()))
+                .andExpect(jsonPath("$[0].id").value(users.get(0).getId().toString()))
+                .andExpect(jsonPath("$[0].username").value(users.get(0).getUsername()))
+                .andExpect(jsonPath("$[0].firstName").value(users.get(0).getFirstName()))
+                .andExpect(jsonPath("$[0].lastName").value(users.get(0).getLastName()))
+                .andExpect(jsonPath("$[0].avatar").value(users.get(0).getAvatar()))
+                .andExpect(jsonPath("$[0].roleId").value(users.get(0).getRoleId().toString()));
+
+        verify(userService).findAll();
+    }
+
+    @Test
+    void shouldFindById_OK() throws Exception {
+        final var user = buildUser();
+
+        when(userService.findUserById(user.getId())).thenReturn(user);
+
+        mvc.perform(MockMvcRequestBuilders.get(BASE_URL + "/" + user.getId()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(user.getId().toString()))
+                .andExpect(jsonPath("$.username").value(user.getUsername()))
+                .andExpect(jsonPath("$.firstName").value(user.getFirstName()))
+                .andExpect(jsonPath("$.lastName").value(user.getLastName()))
+                .andExpect(jsonPath("$.avatar").value(user.getAvatar()))
+                .andExpect(jsonPath("$.roleId").value(user.getRoleId().toString()));
+
+        verify(userService).findUserById(user.getId());
+    }
+
+    @Test
+    void shouldCreateUser_OK() throws Exception{
```

Comment:
Don't forget to rename them all after renaming method from implementation

```suggestion
    void shouldCreate_OK() throws Exception{
```